hw/ip/otbn/dv/otbnsim/sim/sim.py

```diff
@@ -0,0 +1,84 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import struct
+from typing import List
+
+from riscvmodel.model import TerminateException  # type: ignore
+
+from .isa import OTBNInsn
+from .model import OTBNModel
+
+
+class OTBNSim:
+    def __init__(self, model: OTBNModel):
+        self.model = model
+        self.program = []  # type: List[OTBNInsn]
+
+    def load_program(self, program: List[OTBNInsn]) -> None:
+        self.program = program.copy()
+
+    def load_data(self, data: bytes) -> None:
+        # Explicitly zero-pad data to a multiple of 4 bytes if necessary
+        if len(data) & 3:
+            data += b'0' * (4 - (len(data) & 3))
+
+        mem = self.model.state.memory.memory
+        for word_idx, int_val in enumerate(struct.iter_unpack('<I', data)):
+            mem[word_idx] = int_val
```

Comment:
Oops, thanks! Turns out that mypy didn't spot that one, because it can't see inside the `riscvmodel` library's state object (so `mem` has type `Any`). Eventually, we should probably either add typing annotations to that library, or remove the remaining dependencies on it.

Amusingly, this code ran just fine. Presumably, none of my tests actually *read* that memory... :-)